source/common/http/conn_manager_impl.cc

```diff
@@ -1610,6 +1612,16 @@ void ConnectionManagerImpl::ActiveStream::recreateStream(
   ResponseEncoder* response_encoder = response_encoder_;
   response_encoder_ = nullptr;
 
+  Buffer::InstancePtr request_data = std::make_unique<Buffer::OwnedImpl>();
+  bool proxy_body = connection_manager_.enable_internal_redirects_with_body_;
+  if (proxy_body) {
+    const auto& buffered_request_data = filter_manager_.bufferedRequestData();
+    proxy_body = buffered_request_data != nullptr && buffered_request_data->length() > 0;
+    if (proxy_body) {
```

Comment:
Can we just inline all the checks here into one line? Or are we trying to avoid calling bufferedRequestData when the flag is not set? The nested checking of `proxy_body` seems a bit odd when skimming this code

Reply:
updated